include/rive/playable.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef _RIVE_PLAYABLE_HPP_
+#define _RIVE_PLAYABLE_HPP_
+
+#include "rive/animation/loop.hpp"
+#include "rive/math/aabb.hpp"
+#include "rive/math/vec2d.hpp"
+#include <string>
+
+namespace rive {
+    class ArtboardInstance;
+    class Renderer;
+
+    class SMIInput;
+    class SMIBool;
+    class SMINumber;
+    class SMITrigger;
+
+    class Playable {    
+    protected:
+        ArtboardInstance* m_ArtboardInstance;
+
+        Playable(ArtboardInstance*);
+
+    public:
+        virtual ~Playable() {}
+    
+        float width() const;
+        float height() const;
+        AABB bounds() const { return {0, 0, this->width(), this->height()}; }
+
+        virtual std::string name() const = 0;
+
+        virtual Loop loop() const = 0;
+        // Returns true iff the playable is known to not be fully opaque
+        virtual bool isTranslucent() const = 0;
+        // returns -1 for continuous
+        virtual float durationSeconds() const = 0;
+    
+        // If applicaable, resets the internal time to the start of the animation
+        virtual void restart() = 0;
+        // returns true if draw() should be called
+        virtual bool advanceAndApply(float elapsedSeconds) = 0;
```

Comment:
Eventually we should drive towards this being exposed as simply "advance". I get why it's currently advanceAndApply, but eventually that shouldn't matter to the user/embedder.